libosmscout/include/osmscout/util/utf8helper.h

```diff
@@ -35,6 +38,209 @@ extern std::string UTF8Capitalize(const std::string& text);
 
 extern std::string UTF8Transliterate(const std::string& text);
 
+/**
+ * @brief functor implements desired transformation of the character
+ * It has 2 arguments:
+ * - The character struct matching the current code point
+ * - The context, it is the category of the preceding sequence
+ * It must return the new code point to be dumped instead, or NullCodepoint to
+ * discard the sequence.
+ */
+using transform = codepoint (*)(const character*, int context);
+
+extern codepoint transform_nop(const character*, int);
+extern codepoint transform_upper(const character*, int);
+extern codepoint transform_lower(const character*, int);
+extern codepoint transform_capitalize(const character*, int);
+extern codepoint transform_normalize(const character*, int);
+extern codepoint transform_transliterate(const character*, int);
```

Comment:
Just nitpicking - in library is usually used CamelCase for methods and class names with first big letter.